### PR TITLE
chore: upgrade to oidc v2 release

### PR DESCRIPTION
**Pull request by muhlemmer**
This PR targets oidc to the latest refactored version. This is still a WIP to assess the impact untill we release oidc v2 as stable.

```[tasklist]
### Definition of Ready
- [ ] I am happy with the code
- [ ] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] No debug or dead code
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.
```

### CODE REVIEW

**Comment by muhlemmer** [comment]
https://github.com/zitadel/zitadel/actions/runs/4466639918/jobs/7845080060?pr=5437#step:8:26

The complexity isn't changed by this PR. Only the setters are updated:

https://github.com/zitadel/zitadel/pull/5437/files#diff-8c6e576b1abab086a19bd22397882c2d7c146e92d51015a55ddefe29b5be753fR257-R297

Can we ignore this linting error? Because the only way to fix this is to refactor the specific function and this is beyond the scope of the PR.

**Comment by muhlemmer** [comment]
@livio-a . Do not merge. I found a minor bug just now in OIDC, but it's late and I'm going bed. You can review, but there will be a minor version bump of oidc tomorrow. (no further changes in code here).

**Review by livio-a — CHANGES_REQUESTED** [review]
found two issues:
- id tokens do not contain custom claims (`IDTokenClaims.SetUserInfo`) ignores them
- token revocation does not revoke the token(s)

I have a fix for the revocation, let's discuss it on Monday.

File: internal/api/oidc/auth_request.go [hunk 1]
```diff
@@ -203,6 +203,17 @@ func (o *OPStorage) RevokeToken(ctx context.Context, token, userID, clientID str
 	return oidc.ErrServerError().WithParent(err)
 }
 
+func (o *OPStorage) GetRefreshTokenInfo(ctx context.Context, clientID string, token string) (userID string, tokenID string, err error) {
+	refreshToken, err := o.repo.RefreshTokenByID(ctx, token)
+	if err != nil {
+		return "", "", oidc.ErrServerError().WithParent(err)
```
[the remark this answers is not in the document]
Comment (livio-a) [reply internal/api/oidc/auth_request.go]: the correction just overlapped. i fixed it: (https://github.com/zitadel/zitadel/pull/5437/commits/c1f5087c71cb813432460a72b37499a911cec73e#diff-248d1e977ee6e4139976bb970d9d4c9003d0aa36798cf4e5b08c9b7790d2f394R209)

and as the information from that method already returns the user and token ID, which will be passed into the `Revoke` method i changed that too

[the remark this answers is not in the document]
Comment (livio-a) [reply internal/api/oidc/auth_request.go]: thanks @lefelys for checking BTW 😃

**Comment by livio-a** [comment]
@stebenz or @adlerhurst can either of you please check my changes, please? the rest LGTM

**Review by adlerhurst — APPROVED** [review]

**Review by livio-a — APPROVED** [review]

**Comment by livio-a** [comment]
review, testing and fixes took me 3 hours

**Pull request merged**